### DISCUSSION TITLE
Half/Third Width Devices

### DISCUSSION THREAD

**Discussion by kopacko** [netbox-community/netbox#9542] [General]
Hello,

I am preparing to deploy some switches that do not consume a full 1U width across 10 locations.

Is there a way to handle multiple devices within a single 1U location? For example Dell S5212 switches.

![image](https://user-images.githubusercontent.com/1976185/173866992-a12c3385-b7a7-4ba7-a4e5-d0b52bb0c716.png)

These can be racked in pairs within a single 1U space.

**Comment by mtinberg** [comment]
You could create a bogus Parent Device to contain them with two Device Bays and model them as 0U Child devices that slot in to the Parent Device, then you put the bogus Parent device in the rack unit and it allows you to slot in 2 of these Child devices in that space.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Jason Kopacko ***@***.***>
Sent: Wednesday, June 15, 2022 10:32 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Half/Third Width Devices (Discussion #9542)


Hello,

I am preparing to deploy some switches that do not consume a full 1U width across 10 locations.

Is there a way to handle multiple devices within a single 1U location? For example Dell S5212 switches.

[image]<https://user-images.githubusercontent.com/1976185/173866992-a12c3385-b7a7-4ba7-a4e5-d0b52bb0c716.png>

These can be racked in pairs within a single 1U space.

**Reply by necrofrost** [reply]
This is the correct approach. And I've copy/pasted the 2 devices front images next to each other and made a new Front image (no need for a rear image as the devices are not full depth) and attached it to the parent device

**Comment by kopacko** [comment]
I thought about that @mtinberg . I may have to go that route but it is definitely something that should be modeled correctly at some point.

**Reply by iunderwood** [reply]
It's not an incorrect model, per se.  The parent just happens to be the 1U shelf with two device bays that hold the production equipment.  It's just that the housing is infrastructure that is usually just assumed.

**Reply by kopacko** [reply]
Yea I know, I don't think it is an incorrect model. It's just more of, as things get smaller, multiple devices per U are going to happen more and more.

**Comment by mtinberg** [comment]
> should be modeled correctly at some point

I'm not a developer for Netbox, but at some point you enter the land of diminishing returns and the trade off between complexity in the code and db schema and absolute accuracy of what you are trying to model gets unbalanced and that affects everyone who uses Netbox for big or small jobs.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by kopacko** [comment]
Yea I hear ya. It's not a matter of re-inventing the wheel :-) I didn't make these devices, just have to document them.

**Comment by ybizeul** [comment]
Implementing such devices as modules kind of work, up to the point you want to assign IP addresses. You cannot set a Primary IP address on a module, but in that case, it would make sense as the parent is merely a metal plate and not an active chassis/device.

**Reply by candlerb** [reply]
There's a difference between "modules" and "child devices".

A "child device" (which fits in a "device bay" in a "parent device") is a completely independent device, with its own name, its own primary IP address, its own interfaces etc. This is the most appropriate way to model multiple devices which inhabit the same rack unit side-by-side, or blade servers in a blade chassis.

"Modules" expand the capabilities of a device, e.g. by adding additional interfaces.  They are the most appropriate way to model line cards.

In the original posting, I would use a Parent Device to model the special bracket that Dell supply, which has two "device bays", and the Dell S5212 switches would be "child devices" of the bracket.

**Reply by ybizeul** [reply]
Now I feel stupid. Indeed having a "Rack Mount Plate" device with two "Switches" child devices perfectly does the job !
